components/channel_view/channel_view.jsx

```diff
@@ -48,6 +60,46 @@ export default class ChannelView extends React.PureComponent {
         );
     }
 
+    updateChannel(props) {
+        let channel;
+        const fakechannel = (new URLSearchParams(props.location.search)).get('fakechannel');
+        if (fakechannel) {
+            channel = JSON.parse(fakechannel);
+        } else {
+            channel = ChannelStore.getByName(props.match.params.channel);
+
+            if (channel && channel.type === Constants.DM_CHANNEL) {
+                loadNewDMIfNeeded(channel.id);
+            } else if (channel && channel.type === Constants.GM_CHANNEL) {
+                loadNewGMIfNeeded(channel.id);
+            }
+
+            if (channel) {
+                GlobalActions.emitChannelClickEvent(channel);
+            } else {
+                joinChannel(UserStore.getCurrentId(), TeamStore.getCurrentId(), null, props.match.params.channel)(dispatch, getState).then(
+                    (result) => {
+                        if (result.data) {
+                            channel = result.data.channel;
+                            if (channel && channel.type === Constants.DM_CHANNEL) {
+                                loadNewDMIfNeeded(channel.id);
+                            } else if (channel && channel.type === Constants.GM_CHANNEL) {
+                                loadNewGMIfNeeded(channel.id);
+                            }
+                            GlobalActions.emitChannelClickEvent(channel);
+                        } else if (result.error) {
+                            if (props.match.params.team) {
+                                props.history.push('/' + props.match.params.team + '/channels/town-square');
+                            } else {
+                                props.history.push('/');
+                            }
+                        }
+                    }
+                );
+            }
+        }
+    }
```

Comment:
I avoided changing any code that didn't need to be changed. This code was moved from the routing code. I can do that but I want to do it separately. https://mattermost.atlassian.net/browse/ABC-98

Reply:
Ok, it just feels like a step back because we just migrated a bunch of these components away from relying on the flux stores and I'd rather not reintroduce them

Reply:
Not really because they where already secretly relying on the old stores because of the code in the router.

Reply:
Yeah I get it, just feels like it. I'm fine with doing it later in a follow-up ticket